javav2/example_code/ec2/src/test/java/EC2Test.java

```diff
@@ -25,330 +27,265 @@
-        System.out.println("You can SSH to the instance using this command:");
-        System.out.println("ssh -i " + fileNameSc + "ec2-user@" + ipAddress);
-        System.out.println(EC2Scenario.DASHES);
-
-        System.out.println(EC2Scenario.DASHES);
-        System.out.println("10.  Stop the instance.");
-        EC2Scenario.stopInstance(ec2, newInstanceId);
-        System.out.println(EC2Scenario.DASHES);
-
-        System.out.println(EC2Scenario.DASHES);
-        System.out.println("11.  Start the instance.");
-        EC2Scenario.startInstance(ec2, newInstanceId);
-        ipAddress = EC2Scenario.describeEC2Instances(ec2, newInstanceId);
-        assertFalse(ipAddress.isEmpty());
-        System.out.println("You can SSH to the instance using this command:");
-        System.out.println("ssh -i " + fileNameSc + "ec2-user@" + ipAddress);
-        System.out.println(EC2Scenario.DASHES);
-
-        System.out.println(EC2Scenario.DASHES);
-        System.out.println("12. Allocate an Elastic IP address and associate it with the instance.");
-        String allocationId = EC2Scenario.allocateAddress(ec2);
-        assertFalse(allocationId.isEmpty());
-        System.out.println("The allocation Id value is " + allocationId);
-        String associationId = EC2Scenario.associateAddress(ec2, newInstanceId, allocationId);
-        System.out.println("The association Id value is " + associationId);
-        System.out.println(EC2Scenario.DASHES);
-
-        System.out.println(EC2Scenario.DASHES);
-        System.out.println("13. Describe the instance again.");
-        ipAddress = EC2Scenario.describeEC2Instances(ec2, newInstanceId);
-        assertFalse(ipAddress.isEmpty());
-        System.out.println("You can SSH to the instance using this command:");
-        System.out.println("ssh -i " + fileNameSc + "ec2-user@" + ipAddress);
-        System.out.println(EC2Scenario.DASHES);
-
-        System.out.println(EC2Scenario.DASHES);
-        System.out.println("14. Disassociate and release the Elastic IP address.");
-        EC2Scenario.disassociateAddress(ec2, associationId);
-        EC2Scenario.releaseEC2Address(ec2, allocationId);
-        System.out.println(EC2Scenario.DASHES);
-
-
-        System.out.println(EC2Scenario.DASHES);
-        System.out.println("15. Terminate the instance.");
-        EC2Scenario.terminateEC2(ec2, newInstanceId);
-        System.out.println(EC2Scenario.DASHES);
-
-        System.out.println(EC2Scenario.DASHES);
-        System.out.println("16. Delete the security group.");
-        EC2Scenario.deleteEC2SecGroup(ec2, groupId);
-        System.out.println(EC2Scenario.DASHES);
-
-        System.out.println(EC2Scenario.DASHES);
-        System.out.println("17. Delete the keys.");
-        EC2Scenario.deleteKeys(ec2, keyNameSc);
-        System.out.println(EC2Scenario.DASHES);
-        System.out.println("\n Test 16 passed");
+    @Order(8)
+    public void terminateInstance() {

```

Comment:
What happens if the createKeyPair test fails before this test is run. Will the instance still be running?

Reply:
Nope.

Reply:
Can you provide some more detail? Does Junit run every test even if previous ones fail?